src/libs/actions/Policy.ts

```diff
@@ -2574,6 +2574,59 @@ function createPolicyCategory(policyID: string, categoryName: string) {
     API.write(WRITE_COMMANDS.CREATE_WORKSPACE_CATEGORIES, parameters, onyxData);
 }
 
+function renamePolicyCategory(policyID: string, policyCategory: {oldName: string; newName: string}) {
+    const policyCategoryToUpdate = allPolicyCategories?.[`${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyID}`]?.[policyCategory.oldName] ?? {};
+
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyID}`,
+                value: {
+                    [policyCategory.oldName]: {
+                        name: policyCategory.newName,
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyID}`,
+                value: {
+                    errors: {
+                        [policyCategory.oldName]: null,
+                        [policyCategory.newName]: {
+                            ...policyCategoryToUpdate,
+                            name: policyCategory.newName,
+                        },
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_CATEGORIES}${policyID}`,
+                value: {
+                    [policyCategory.oldName]: {
+                        name: policyCategory.oldName,
+                        errors: ErrorUtils.getMicroSecondOnyxError('workspace.categories.genericFailureMessage'),
+                        pendingAction: null,
+                    },
+                },
+            },
+        ],
+    };
+
+    const parameters = {
+        policyID,
+        categories: JSON.stringify([{[policyCategory.oldName]: policyCategory.newName}]),
```

Comment:
@ArekChr sorry, the API type for the `categories` param was wrong. We just send an object with key = oldName and value = newName

```suggestion
        categories: JSON.stringify({[policyCategory.oldName]: policyCategory.newName}),
```

The change above should fix the issue you're seeing. Another issue that I saw is that after renaming the category, we're navigated back to the edit page and then a not found page shows up since the category was renamed and the `/workspace/<policyID>/categories/<oldCategoryName>` route no longer exists, we should replace it with `/workspace/<policyID>/categories/<newCategoryName>`

Reply:
Do we have any similar cases where we manipulate the URL while editing names? This bug is tricky one because I'm unable to access the success promise from the 'edit category' operation. Additionally, our navigation setup prevents the use of 'navigation replace'. Consequently, I've closed RHP to mitigate the issue.